tool/tsh/daemonstop_windows.go

```diff
@@ -0,0 +1,90 @@
+//go:build windows
+// +build windows
+
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"syscall"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/sys/windows"
+)
+
+var (
+	k32           = syscall.NewLazyDLL("kernel32.dll")
+	freeConsole   = k32.NewProc("FreeConsole")
+	attachConsole = k32.NewProc("AttachConsole")
+)
+
+// onDaemonStop implements the "tsh daemon stop" command. It handles graceful shutdown of the daemon.
```

Comment:
You might be able to do without this. Go's signal package already handles some of this Windows behavior for you: https://pkg.go.dev/os/signal#hdr-Windows